dataactvalidator/filestreaming/loaderUtils.py

```diff
@@ -25,85 +28,62 @@ def compareRecords (recordA,recordB, fields) :
-                        # Create empty dict for this field
-                        valuePresent[fieldMap[field]] = {}
-                else:
-                    raise KeyError("".join(["Found unexpected field ", str(field)]))
-            # Check that all fields are present
-            for field in fieldMap:
-                if not field in fieldNames:
-                    raise ValueError("".join([str(field)," is required for loading table ", str(type(model))]))
-            # Open DictReader with attribute names
-            reader = csv.DictReader(csvfile,fieldnames = attributeNames)
-            # For each row, create instance of model and add it
-            for row in reader:
-                skipInsert = False
-                for field in fieldOptions:
-                    if row[field] is None:
-                        # If field is empty set to an empty string
-                        row[field] = ""
-                    # For each field with options present, modify according to those options
-                    options = fieldOptions[field]
-                    if "strip_commas" in options:
-                        # Remove commas from numeric fields
-                        row[field] = row[field].replace(",","")
-                        if row[field] == "":
-                            # If empty, set to 0
-                            row[field] = "0"
-                    if "pad_to_length" in options:
-                        padLength = options["pad_to_length"]
-                        row[field] = Validator.padToLength(row[field],padLength)
-                    if "skip_duplicates" in options:
-                        if row[field] is None or len(row[field].strip()) == 0 or row[field] in valuePresent[field]:
-                            # Value not provided or already exists, skip it
-                            skipInsert = True
-                        else:
-                            # Insert new value
-                            valuePresent[field][row[field]] = True
-                record = model(**row)
-                if not skipInsert:
-                    try:
-                        interface.session.merge(record)
-                        interface.session.commit()
+        # toss out blank rows
+        data.dropna(inplace=True, how='all')
+
+        # Fix up cells that have spaces instead of being empty.
+        # Set the truly empty cells to None so they get inserted to db as NULL
+        # TODO: very ugly function below...is there a better way?
+        data.applymap(lambda x: str(x).strip() if len(str(x).strip()) else None)
+
+        # clean the dataframe column names
+        data.rename(columns=lambda x: cls.cleanColNamesFunction(x), inplace=True)
+        # make sure all values in fieldMap parameter are in the dataframe/csv file
+        for field in fieldMap:
+            if field not in list(data.columns):
+                raise ValueError("{} is required for loading table{}".format(field, model))
```

Comment:
Noting that this is a prime example for why the `fieldMap` approach is problematic. Rather than relying of methods/functions with named arguments to verify the parameters at call-time (and potentially before, using static analysis), we're throwing an exception half way through this function.